### PR TITLE
TechDebt: Map container refactor

### PR DESCRIPTION
**Pull request by al-rosenthal**
## Links to Jira Tickets

[SIMSBIOHUB-351](https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-351)

## Description of Changes

- Updated `DrawControls.tsx`
- Removed old `MapComponent.tsx`
- Removed old `MapBoundary.tsx`
- Removed project location code from App/ Api
- Updated OpenApi schema to not require locations on project updated/ create
- Updated Survey Study area details to not require a map component
- Removed old search page

## Testing Notes

- You should be able to create and edit projects as normal
- You should be able to create a survey with locations (map should still be working, draw controls should be functional)
- Add a marked animal with a capture event (map should still be working)
- Add a telemetry device to that animal, open the database and change the `critterbase_critter_id` to `826aa7b4-31cf-44d5-80cd-fc159e42d7e7`, then view telemetry 
- Open Manage Observations and add a sampling site(map should still be working)
- Add a few observations and navigate back to survey details (map should still be working)

### CODE REVIEW

File: api/src/services/eml-service.ts [hunk 1]
```diff
@@ -873,36 +872,6 @@ export class EmlService extends DBService {
...
   }
 
-  /**
-   * Creates an object representing geographic coverage pertaining to the given project
-   *
-   * @param {IGetProject} projectData
-   * @return {*}  {Record<string, any>}
-   * @memberof EmlService
-   */
-  _getProjectGeographicCoverage(projectData: IGetProject): Record<string, any> {
```
Comment (curtisupshall) [inline api/src/services/eml-service.ts:883]: Even though we don't use the EML service, if there's a quick + easy way to replace this method with something similar, we should probably do it. Maybe produce a bounding box covering a concatenation of all of the geometries for all of the surveys belonging to this project? Might want to ask Macgreggor

**Comment by NickPhura** [comment]
A few other remnants of project location that I think can be deleted as well:
- `location_description`
- `IGetProjectForViewResponseLocation`
- `PutLocationData`

The `DrawControls2.tsx` and `StudyAreaMap.tsx` files can be deleted.

Its unfortunately not obvious, but anywhere the `fullscreenControl={true}` is used, you have to import the below, otherwise the full screen button doesn't render:
```
import 'leaflet-fullscreen/dist/leaflet.fullscreen.css';
import 'leaflet-fullscreen/dist/Leaflet.fullscreen.js';
```

File: app/src/features/projects/edit/EditProjectForm.tsx
Comment (curtisupshall) [inline app/src/features/projects/edit/EditProjectForm.tsx:1]: You forgot a mention of `location` in `EditProjectPage.tsx` (Ctrl+Shift+F for "ENTITIES.location")

File: api/src/models/project-view.ts
Comment (curtisupshall) [inline api/src/models/project-view.ts:1]: You forgot `project-update.ts` and `project-update.test.tsx`

File: api/src/paths/project/{projectId}/update.ts
Comment (curtisupshall) [inline api/src/paths/project/{projectId}/update.ts:1]: This file still contains mention of `location` (as well as its test file)

File: api/src/repositories/project-repository.ts
Comment (curtisupshall) [inline api/src/repositories/project-repository.ts:1]: This file still contains mention of `location` (and its test file)

File: app/src/utils/mapUtils.ts
Comment (curtisupshall) [inline app/src/utils/mapUtils.ts:1]: +1 for map utils 👍

**Review by curtisupshall — DISMISSED** [review]
Code looks good. There are still a handful of mentions of project location.

On another note, this PR seems to remove the Search Page from the app, but the changelist doesn't mention this (unless I'm missing something). If this is tracked in a ticket, reference the ticket (or maybe add it as a requirement in an existing ticket), but otherwise just add it to the list of changes in the PR

**Review by curtisupshall — COMMENTED** [review]
Sorry. Approved by mistake

File: api/src/services/eml-service.ts (hunk 1 above)
Comment (NickPhura) [reply api/src/services/eml-service.ts:883]: Could just replace this call (wherever its used) with a call to `_getSurveyGeographicCoverage`. They return the exact same shape object.

File: app/src/features/projects/edit/EditProjectForm.tsx
Comment (al-rosenthal) [reply app/src/features/projects/edit/EditProjectForm.tsx:1]: removed

File: api/src/models/project-view.ts
Comment (al-rosenthal) [reply api/src/models/project-view.ts:1]: removed

File: api/src/paths/project/{projectId}/update.ts
Comment (al-rosenthal) [reply api/src/paths/project/{projectId}/update.ts:1]: removed

File: api/src/repositories/project-repository.ts
Comment (al-rosenthal) [reply api/src/repositories/project-repository.ts:1]: removed

File: api/src/services/eml-service.ts (hunk 1 above)
Comment (curtisupshall) [reply api/src/services/eml-service.ts:883]: Ahh good catch

Comment (al-rosenthal) [reply api/src/services/eml-service.ts:883]: I've updated this part of the code to combine all the associated surveys into a single coverage object

**Review by NickPhura — APPROVED** [review]

**Review by curtisupshall — APPROVED** [review]
Code looks good to me

**Review by curtisupshall — APPROVED** [review]
Tested this manually, appears that everything is functioning normally.

**Pull request merged**